libbeat/docs/outputconfig.asciidoc

```diff
@@ -86,13 +86,51 @@ output.elasticsearch:
   protocol: "https"
   username: "{beatname_lc}_internal"
   password: "{pwd}"
-
 ------------------------------------------------------------------------------
 
 
 For more information about securing {beatname_uc}, see
 <<securing-{beatname_lc}>>.
 
+
+To configure the Elasticsearch output with https://www.elastic.co/guide/en/elasticsearch/reference/6.x/getting-started-index-lifecycle-management.html[index lifecycle management] feature configure it as following:
+
+[source,yaml]
+------------------------------------------------------------------------------
+output.elasticsearch:
+  hosts: ["localhost:9200"]
+  ilm.enabled: true
+------------------------------------------------------------------------------
+
+This will overwrite your index settings and adjust the Beats template to contain the correct settings.
+
+Note: If you have loaded a template before into Elasticsearch with the same version you must overwrite the template.
+Note: This feature only works with X-Pack Basic.
+
+The default `rollover_alias` used is `{beatname_lc}-{beat.version}`. It's not possible to remove the `{beat.version}`
+from the rollover_alias name but the prefix part can be adjusted. This will also directly affect the name of the
+rollover indices. The default pattern used for the rollover index is `00001`. It can be adjusted with the
+`ilm.pattern` config. See config below for an example:
+
+[source,yaml]
+------------------------------------------------------------------------------
+output.elasticsearch:
+  hosts: ["localhost"]
+  ilm.enabled: true
+  ilm.rollover_alias: "metricbeat"
+  ilm.pattern: "00001"
```

Comment:
ES will always use 6 digits when rolling over. Also - when do you expect people to change this pattern and if so to what end?

Reply:
Based on the comment from @colings86 it is 5: https://github.com/elastic/beats/pull/7963#discussion_r210322415 Did this change?

For changing it: My initial idea was that users could use here also date math: https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-rollover-index.html#_using_date_math_with_the_rollover_api We decided to do that by default but at the moment it causes some issues as it as a `/` in the date math and escaping does not work properly. It's not an issue on the ILM side but on how requests are processed on the Beats side.

Reply:
This was bad information from me sorry. It should be 6 digits total

Reply:
pattern updated to 000001